filterdb/db.go

```diff
@@ -155,30 +167,22 @@ func putFilter(bucket walletdb.ReadWriteBucket, hash *chainhash.Hash,
 // storage.
 //
 // NOTE: This method is a part of the FilterDatabase interface.
-func (f *FilterStore) PutFilter(hash *chainhash.Hash,
-	filter *gcs.Filter, fType FilterType) error {
-
+func (f *FilterStore) PutFilter(filterData *FilterData) error {
 	return walletdb.Update(f.db, func(tx walletdb.ReadWriteTx) error {
 		filters := tx.ReadWriteBucket(filterBucket)
 
 		var targetBucket walletdb.ReadWriteBucket
-		switch fType {
+		switch filterData.Type {
```

Comment:
I think we can also start to use `Batch` here as well instead of `Update`.

Reply:
cool - added this in the next commit 👍